packages/cli/src/commands/element.ts

```diff
@@ -514,6 +515,23 @@ const safeGetElementId = (maybeElementIdPath: string, output: CliOutput): ElemID
   }
 }
 
+export const getParentUrl = async (workspace: Workspace, childElement: Element): Promise<string | undefined> => {
+  const parentsArray = getParents(childElement)
+  if (parentsArray.length === 0) {
+    return undefined
+  }
+  const parentRef = await awu(parentsArray).find(async parent => {
+    const parentElement = await workspace.getValue(parent.elemID)
+    if (!isElement(parentElement)) {
+      return false
+    }
+    return parentElement.annotations[CORE_ANNOTATIONS.SERVICE_URL] !== undefined
+  })
+
+  const parentElem = parentRef !== undefined ? await workspace.getValue(parentRef.elemID) : undefined
+  return isElement(parentElem) ? parentElem.annotations[CORE_ANNOTATIONS.SERVICE_URL] : undefined
```

Comment:
```suggestion
  const url = await awu(parentsArray)
    .map(async parentEntry => {
      const parent = await workspace.getValue(parentEntry.elemID)
      return parent.annotations[CORE_ANNOTATIONS.SERVICE_URL]
    })
    .find(serviceUrl => {
      return serviceUrl !== undefined
    })
  return url
```
the point of doing it in that way was to not use the "expenisive" func of getValue more than we need, so let try to think of a way that we dont need to call it again after the iteration.
this is only a suggestion of direction

Reply:
ill try to figure it out
i tried some ways to avoid the calliing of the getValue function again after the iteration
but had some issues trying to do so